enhancements/cluster-logging/hypershift-audit-logs.md

```diff
@@ -0,0 +1,219 @@
+and validation may be needed:
+- validate that the forwarding request is valid and allowed.
+- possibly replace named outputs with a suitable output configuration;
+  if we want to restrict the user to a fixed set of alternatives
+- possibly add credentials derived from the guest cluster or the management cluster.
+  
+### Risks and Mitigations
+
+CPU/Network cost of extra log forwarding
+- We have to do this anyway, but user might choose a more verbose policy.
+- We could restrict the available logs at the API server to control this.
+
+### Drawbacks
+
+## Design Details
+
+### Open Questions
+
+- Is there any way to write the audit logs to a file system that the log forwarder can read?
+  This would be simpler & more reliable than a web hook.
+- Can an operator on the management cluster reconcile and update status on resources in the guest cluster directly?
+  The API server is running in the HCP on the management cluster, but need to check for possible snags.
+- Should we allow the user to specify any destination for logs from the management cluster? If not:
+   - Why not? We allow that freedom on self contained clusters. I see no new security/privacy issues.
+   - We can instead provide a fixed set of named alternatives. The `HyperShiftLoggingOperator`
+     can translate those names into known configurations.
+- How do we manage credentials? Assuming they are stored in the HCP API server, can the management cluster
+  access secrets for TLS security and the like? If not how do we get at them.
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+#### Tech Preview -> GA
+
+#### Removing a deprecated feature
+
+None.
+
+### Upgrade / Downgrade Strategy
+
+Nothing new.
+
+### Version Skew Strategy
+
+Nothing new.
+
+### Operational Aspects of API Extensions
+
+An operator on the management server that watches objects on the guest cluster is new (is it?)
+Since the guest API server runs in the HCP on the management server, it seems this should be possible,
+but there may be some hidden snags.
+
+This proposal requires 2 new operators to run on management clusters:
+CLO and HyperShiftLoggingOperator.
+The custom Splunk forwarder will be removed.
+
+#### Failure Modes
+
+- Invalid `HyperShiftLogForwarder.spec`: Indicate in `HyperShiftLogForwarder.status`
+- Invalid policy reference or policy: Indicate in `HyperShiftLogForwarder.status`
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+None.
+
+## Alternatives
+
+The current work around solves the immediate problem (Cloudwatch yes-or-no), but is not a good long term solution:
+- Requires a call to support to enable, can't be controlled from guest cluster.
+- Custom Cloudwatch client code hastily added to splunk exporter duplicates a CLO feature.
+- Exporter was not intended to be a client, just a log filter.
+- SD splunk forwarder duplicates Splunk output in the CLO.
```

Comment:
But adds the ability to filter, which is critical for our use case.

Reply:
Agreed - and to customer use cases also. Hence the goal of transplanting it to the CLO where all can benefit.